nltk/internals.py

```diff
@@ -108,12 +108,13 @@ def java(cmd, classpath=None, stdin=None, stdout=None, stderr=None, blocking=Tru
 
     :raise OSError: If the java command returns a nonzero return code.
     """
-    if stdin == 'pipe':
```

Comment:
It might worth updating the docstring, as it seems to me that the code and docs are not in sync. Specifically, the docs above are saying:
```
Valid values are ``subprocess.PIPE``,
an existing file descriptor (a positive integer), an existing
file object, and None.
```

But the function is happy to take strings in the form of `"pipe"`, which is not mentioned there.

Reply:
`    :param stdin, stdout, stderr: Specify the executed programs'
        standard input, standard output and standard error file
        handles, respectively.  Valid values are ``subprocess.PIPE``,
        an existing file descriptor (a positive integer), an existing
        file object, 'pipe', 'stdout', 'devnull' and None.  ``subprocess.PIPE`` indicates that a
        new pipe to the child should be created.  With None, no
        redirection will occur; the child's file handles will be
        inherited from the parent.  Additionally, stderr can be
        ``subprocess.STDOUT``, which indicates that the stderr data
        from the applications should be captured into the same file
        handle as for stdout.`

Is this what you had in mind?

Reply:
👍